### PR TITLE
Feat/xdsl modem tasks

### PR DESCRIPTION
**Pull request by Steffy29**
add tasks tab on xDSL detail to display the tasks and their status for the user.

ref: XDSL-5047

need to validate the PR https://github.com/ovh-ux/ovh-api-services/pull/105 before deployment to access the list of tasks and task detail for xDSL.

- [x] ovh-ux/ovh-api-services#105

### CODE REVIEW

File: client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html [hunk 1]
```diff
@@ -1,24 +1,5 @@
...
-                </oui-spinner>
-                <span data-translate="xdsl_modem_firewall_doing"></span>
-            </span>
-        <span data-ng-if="!FirewallCtrl.mediator.tasks.changeModemConfigEasyFirewallLevel" data-translate="xdsl_modem_firewall_submit"></span>
-    </button>
-</form>
+<div class="form-group">
+    <input type="text" class="form-control"
+           id="configFirewall"
+           name="configFirewall" value="{{FirewallCtrl.firewallCurrentLevel.label | translate}}" disabled>
```
Comment (frenautvh) [inline client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html]: you can also do ```ng-value="FirewallCtrl.firewallCurrentLevel.label | translate"```

File: client/app/telecom/pack/xdsl/pack-xdsl.html [hunk 2]
```diff
@@ -36,6 +36,10 @@
                           class="fa fa-plus"
                           aria-hidden="true"></span>
                 </responsive-tab-more>
+                <responsive-tab data-state="telecom.pack.xdsl.tasks"
+                                data-dropdown-title="{{ 'xdsl_tabs_tasks' | translate }}">
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/pack-xdsl.html]: not possible here

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <toast-message></toast-message>
+
+        <div class="widget-presentation">
+            <h2 class="widget-presentation-title"
+                data-translate="xdsl_tasks_list">
+            </h2>
+
+
+            <oui-datagrid data-rows="XdslTasks.taskIds" data-row-loader="XdslTasks.transformItem($row)">
+                <oui-column data-title="'xdsl_tasks_name' | translate" data-property="function" data-sortable="asc"></oui-column>
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html]: not possible here

[the remark this answers is not in the document]
Comment (frenautvh) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html]: ```data-title=":: 'xdsl_tasks_name' | translate"``` should be fine no?

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+angular.module('managerApp')
+  .controller('XdslTasksCtrl', function ($scope, $stateParams, $translate, $q, PAGINATION_PER_PAGE, OvhApiXdsl, Toast) {
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js]: done

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    <div class="telecom-section-content">
+
+        <toast-message></toast-message>
+
+        <div class="widget-presentation">
+            <h2 class="widget-presentation-title"
+                data-translate="xdsl_tasks_list">
+            </h2>
+
+
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html]: done

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        (taskIds) => {
+          self.taskIds = taskIds.map(taskId => ({ id: taskId }));
+        },
+        (error) => {
+          Toast.error([$translate.instant('an_error_occured'), error.data.message].join(' '));
+        },
+      );
+    };
+
+    self.transformItem = function (row) {
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js]: done

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    self.tasks = [];
+    self.serviceName = $stateParams.serviceName;
+    self.filter = {
+      perPage: PAGINATION_PER_PAGE,
+    };
+
+    self.$onInit = function () {
+      $q.all([
+        self.getTasks(),
+      ]);
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.controller.js]: done

**Comment by ovh-cds** [comment]
CDS Report build-manager#106.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✘

File: client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html (hunk 1 above)
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html]: done

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html]: done

**Comment by ovh-cds** [comment]
CDS Report build-manager#107.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✘

File: client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html [hunk 8]
```diff
@@ -1,24 +1,5 @@
...
-                </oui-spinner>
-                <span data-translate="xdsl_modem_firewall_doing"></span>
-            </span>
-        <span data-ng-if="!FirewallCtrl.mediator.tasks.changeModemConfigEasyFirewallLevel" data-translate="xdsl_modem_firewall_submit"></span>
-    </button>
-</form>
+<div class="form-group">
+    <input type="text" class="form-control"
+           id="configFirewall"
+           name="configFirewall" ng-value="FirewallCtrl.firewallCurrentLevel.label | translate" disabled>
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html]: Missing `data-` prefer (`ng-value`)

File: client/app/telecom/pack/xdsl/tasks/packs-xdsl-tasks.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+angular.module('managerApp').config(($stateProvider) => {
+  $stateProvider.state('telecom.pack.xdsl.tasks', {
+    url: '/tasks',
+    views: {
+      'xdslView@telecom.pack.xdsl': {
+        templateUrl: 'app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html',
+        controller: 'XdslTasksCtrl',
+        controllerAs: 'XdslTasks',
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/tasks/packs-xdsl-tasks.js]: Why not prefer `$ctrl`? We tend to homogenize that as much as we can:)

File: LICENSE [hunk 10]
```diff
@@ -1,25 +1,29 @@
-Copyright (c) 2013-2018, OVH SAS.
+BSD 3-Clause License
```
Comment (antleblanc) [inline LICENSE]: This file has been updated and fixed on the `master` (synced with `develop`) so I guess you could do a rebase on `develop`.

File: client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html (hunk 8 above)
Comment (antleblanc) [reply client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html]: I updated my previous comment:) Hope it help you.

Comment (Steffy29) [reply client/app/telecom/pack/xdsl/modem/firewall/pack-xdsl-modem-firewall.html]: OK. Great

File: client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+            </h2>
+
+            <oui-datagrid data-rows="XdslTasks.taskIds" data-row-loader="XdslTasks.transformItem($row.id)">
+                <oui-column data-title="::'xdsl_tasks_name' | translate" data-property="function" data-sortable="asc"></oui-column>
+                <oui-column data-title="::'xdsl_tasks_id' | translate" data-property="id" data-sortable></oui-column>
+                <oui-column data-title="::'xdsl_tasks_status' | translate" data-property="status" data-sortable>
+                    <span class="label"
+                          data-ng-class="{
+                                'label-success': $row.status === 'done',
+                                'label-danger': $row.status === 'error' || 'problem',
```
Comment (FredericEspiau) [inline client/app/telecom/pack/xdsl/tasks/pack-xdsl-tasks.html]: This label will always be selected, you meant `$row.status === 'error' || $row.status === 'problem'`, or less lisible but more maintainable `['error', 'problem'].indexOf($row.status) !== -1`

File: client/app/telecom/pack/xdsl/tasks/packs-xdsl-tasks.js (hunk 9 above)
Comment (FredericEspiau) [reply client/app/telecom/pack/xdsl/tasks/packs-xdsl-tasks.js]: It's also the default name for controllers in components and we should tend to using components

File: client/app/telecom/pack/xdsl/modem/wifi/config/pack-xdsl-modem-wifi-config.html [hunk 12]
```diff
@@ -318,7 +318,7 @@
 
                     <!-- Channel -->
                     <div class="form-group"
-                         data-ng-if="ConfigWifiCtrl.wifi.frequency !== '5GHz'">
+                         data-ng-if="ConfigWifiCtrl.wifi.frequency !== '5GHz' && ConfigWifiCtrl.wifi.guest === false">
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/pack/xdsl/modem/wifi/config/pack-xdsl-modem-wifi-config.html]: The need is to disable the possibility to modify channel for wifi guest.

**Pull request merged**